aws/resource_aws_route53_resolver_firewall_rule_group_association.go

```diff
@@ -0,0 +1,217 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/route53resolver"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/service/route53resolver/finder"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/service/route53resolver/waiter"
+)
+
+func resourceAwsRoute53ResolverFirewallRuleGroupAssociation() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsRoute53ResolverFirewallRuleGroupAssociationCreate,
+		Read:   resourceAwsRoute53ResolverFirewallRuleGroupAssociationRead,
+		Update: resourceAwsRoute53ResolverFirewallRuleGroupAssociationUpdate,
+		Delete: resourceAwsRoute53ResolverFirewallRuleGroupAssociationDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateRoute53ResolverName,
+			},
+
+			"firewall_rule_group_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateRoute53ResolverName,
+			},
+
+			"mutation_protection": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.StringInSlice(route53resolver.MutationProtectionStatus_Values(), false),
+			},
+
+			"priority": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+
+			"vpc_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateRoute53ResolverName,
+			},
```

Comment:
Nit: Should this use a different validation?

Reply:
Good spot. I'll remove the validation on this field.

Reply:
Done (plus removed the validation on `firewall_rule_group_id`).